Mage/src/main/java/mage/game/permanent/token/TokenImpl.java

```diff
@@ -212,13 +218,13 @@ private static void putOntoBattlefieldHelper(CreateTokenEvent event, Game game,
                 ((TokenImpl) token).lastAddedTokenId = permanent.getId();
             }
             game.addSimultaneousEvent(new ZoneChangeEvent(permanent, permanent.getControllerId(), Zone.OUTSIDE, Zone.BATTLEFIELD));
-            if (permanent instanceof PermanentToken) {
+            if (permanent instanceof PermanentToken && created) {
                 game.addSimultaneousEvent(new CreatedTokenEvent(event.getSourceId(), (PermanentToken) permanent));
             }
             if (attacking && game.getCombat() != null && game.getActivePlayerId().equals(permanent.getControllerId())) {
                 game.getCombat().addAttackingCreature(permanent.getId(), game, attackedPlayer);
             }
-            if (!game.isSimulation()) {
+            if (created) {
                 game.informPlayers(controller.getLogName() + " creates a " + permanent.getLogName() + " token");
```

Comment:
You must inform users about new tokens all the time, even on copied tokens (just change the text).

Also add rules about `wasCreated` different somewhere in the code:
> The token that a resolving copy of a spell becomes isn’t said to have been “created.” (2020-09-25)